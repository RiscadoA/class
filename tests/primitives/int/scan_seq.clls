proc main() {
    cut {
        send ch(x. scan(x));
        send ch(y. scan(y: lint));
        send ch(z. scan(z));
        close ch
        |ch: ~send lint; send lint; send lint; close|
        recv ch(x); println(x);
        recv ch(y); println(y);
        recv ch(z); println(z);
        wait ch; ()
    }
};;
