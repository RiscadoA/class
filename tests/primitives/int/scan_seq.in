47 42 9 trash
