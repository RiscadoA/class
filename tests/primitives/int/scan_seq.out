47
42
9
